mucommander-core/src/main/java/com/mucommander/core/BrowseLocationThread.java

```diff
@@ -168,6 +168,8 @@ public void run() {
         if(credentialsMapping!=null) {
             newCredentialsMapping = credentialsMapping;
             CredentialsManager.authenticate(folderURL, newCredentialsMapping);
+        } else if (isSftpWithPrivateKey()) {
+            // Special case: for SFTP connection that uses a private key, no need to re-authenticate
```

Comment:
based on the top-level thread, how about changing the else-if below to:
```java
else if(!folderURL.containsCredentials() && !isSftpWithPrivateKey() && ...
```
and we can extend the comment there with: "There is no need to get credentials from users for locations that we already have non-password based credentials for in CredentialsManager, like SFTP with a private key" 

writing the text above made me wonder whether we should rather copy the private key from the URL that is presented in the folder panel to the new URL that is passed here - did you consider this option? (basically saying that if the scheme+host of the presented file matches those of the new URL that is specified, to copy the properties from the former to the latter)

Reply:
I don't mind moving the (negated) condition into the existing `else if`. My though was to not make the existing condition more complex. I like to do it sometimes even if the block itself is empty, and I just use it for a comment explaining why it's a NO-OP. Please LMK if you'd like me to go forward with the change or just change the comment and leave the new `else if` clause as is.

For the other implementation option (`copy the private key from the URL that is presented in the folder panel to the new URL that is passed here`) - actually that was my original idea but it was not trivial to implement. It seems like the private key property was missing, and I didn't want to do extra manipulation to get it from the credential manager, since it already takes care of that internally.

In addition, since we always have a fallback of displaying the auth dialog on auth error, I figured it would be fine to assume that if this is an SFTP connection and PK was used, the credentials should be correct.

Reading the code, it seemed to me like the existing `else if` [block](https://github.com/mucommander/mucommander/blob/5512bbfafff784062784301330b9da15a0c3c743/mucommander-core/src/main/java/com/mucommander/core/BrowseLocationThread.java#L177) was added at some later stage as an optimization and figured I would just skip it.

Reply:
> I don't mind moving the (negated) condition into the existing `else if`. My though was to not make the existing condition more complex. I like to do it sometimes even if the block itself is empty, and I just use it for a comment explaining why it's a NO-OP. Please LMK if you'd like me to go forward with the change or just change the comment and leave the new `else if` clause as is.

ok, yeah - it would complicate the existing condition but overall it would better reflect our intention so please change the existing condition
 
> For the other implementation option (`copy the private key from the URL that is presented in the folder panel to the new URL that is passed here`) - actually that was my original idea but it was not trivial to implement. It seems like the private key property was missing, and I didn't want to do extra manipulation to get it from the credential manager, since it already takes care of that internally.

ack, we can reconsider this when we generalize this logic for other credential properties, that's fine

Reply:
Sounds good - I made the change and tested locally again. We should be good to go!